### PR TITLE
Improve the welcome file handling.

### PR DESCRIPTION
**Pull request by knightburton**
Remove the welcome file from the list if that is completely untouched.
If the user modified that file or added to the sending list then can not be removed.

IoT.jsCode-DCO-1.0-Signed-off-by: Imre Kiss kissi.szeged@partner.samsung.com

### CODE REVIEW

File: src/app/session.js [hunk 1]
```diff
@@ -501,6 +510,15 @@ export default class Session {
 
     this._surface.showEditor();
 
+    // Remove the welcome file if that is untouched.
+    if (this._id.next !== 1 &&
+        this.getFileDataById(1) &&
```
Comment (daeyeon) [inline src/app/session.js]: I suggest using a cached result returned from `this.getFileDataById(1)`, or, creating a new function for checking if the welcome file has no change.

Comment (knightburton) [reply src/app/session.js]: You are right, I will fix this :)

**Comment by knightburton** [comment]
@daeyeon I've updated the PR.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**